geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/concurrent/CachedThreadPool.scala

```diff
@@ -59,12 +60,15 @@ class CachedThreadPool(maxThreads: Int) extends AbstractExecutorService {
   override def isTerminated: Boolean = stopped.get && available == maxThreads // should be safe to read a volatile primitive
 
   override def awaitTermination(timeout: Long, unit: TimeUnit): Boolean = {
-    // TODO GEOMESA-2795 could probably make a wait instead of a poll
-    val end = System.nanoTime() + unit.toNanos(timeout)
-    while (!isTerminated && System.nanoTime() < end) {
-      Thread.sleep(100)
+    lock.lock()
+    try {
+      if (isTerminated) { true } else {
+        done.await(timeout, unit)
```

Comment:
Could the 'done' here be a def/lazy val?  (Probably doesn't matter)

Reply:
i don't think so - it gets called by the last thread that runs so it would always be created. can't be a def b/c it's shared between the waiter and notifier thread.